### PR TITLE
add cmake-format to the pre-commit checks and updates

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add cmake-format to the pre-config list.  FIx issue #1078

### Proposed changes

- add cmake-format pre-commit to the formatting checks

### CODE REVIEW

**Comment by nightlark** [comment]
It looks like it's better than before at not splitting quoted arguments across lines. It seems to mess up comments in several cases.

- commented out code blocks
- `######` section type comments get a space added after the first `#` (a few commented out files also got this
- merging commented lines that were intended to be separate; separate sentences on different lines, or commented out blocks of code (which may be better to remove entirely)

**Comment by phlptp** [comment]
Most of the ####### blocks that we used have were turned into # ----------------------- or something to that effect a while ago, so we could do that to the remaining ones to be consistent.  I don't really think the extra ' ' is that much of an issue.  

The reflow of comments on the code sections is a little more problematic, but the reality is those should probably be removed anyway right now so not likely that big of an issue either.

File: examples/BrokerServerTestCase/CMakeLists.txt [hunk 1]
```diff
@@ -14,13 +14,10 @@ target_link_libraries(brokerServerTestCase PUBLIC HELICS::apps)
...
-    #    NAME brokerServerTestCase
-    #    COMMAND
-    #        ${PROJECT_SOURCE_DIR}/scripts/test_example.sh 10
-    #        $<TARGET_FILE:nonlings_fed1_cpp> $<TARGET_FILE:nonlings_fed2_cpp>
-    #)
-    #set_property(TEST nonlings_example PROPERTY LABELS Continuous)
+    # add_test(
+    # NAME brokerServerTestCase COMMAND ${PROJECT_SOURCE_DIR}/scripts/test_example.sh 10
+    # $<TARGET_FILE:nonlings_fed1_cpp> $<TARGET_FILE:nonlings_fed2_cpp> ) set_property(TEST
+    # nonlings_example PROPERTY LABELS Continuous)
```
Comment (phlptp) [inline examples/BrokerServerTestCase/CMakeLists.txt:20]: @nightlark  is this whole if clause something we need?

Comment (nightlark) [reply examples/BrokerServerTestCase/CMakeLists.txt:20]: I think we can get rid of it.

**Comment by phlptp** [comment]
I made one more tweak that cleaned up a few cases a little better, and got rid of that commented out section.

File: benchmarks/helics/CMakeLists.txt [hunk 2]
```diff
@@ -34,17 +34,17 @@ set(HELICS_MULTINODE_BENCHMARKS
     TimingLeafFederate
 )
 
-if (NOT DEFINED HELICS_BENCHMARK_SHIFT_FACTOR)
+if(NOT DEFINED HELICS_BENCHMARK_SHIFT_FACTOR)
     include(ProcessorCount)
-    ProcessorCount(N)
-    if (N LESS 5)
-       set (HELICS_BENCHMARK_SHIFT_FACTOR -1)
+    processorcount(N)
```
Comment (nightlark) [inline benchmarks/helics/CMakeLists.txt:39]: I guess this is okay since CMake commands are case insensitive.

**Pull request merged**